authorization/src/main/java/uk/gov/di/ipv/cri/common/api/handler/AuthorizationHandler.java

```diff
@@ -69,7 +71,12 @@ public APIGatewayProxyResponseEvent handleRequest(
             // validate
             authorizationValidatorService.validate(authenticationRequest, sessionItem);
 
-            // create authorization
+            // create authorization if not already present
+            if (StringUtils.isBlank(sessionItem.getAuthorizationCode())) {
+                sessionService.createAuthorizationCode(sessionItem);
+                eventProbe.log(INFO, "Authorization code generated successfully.");
```

Comment:
change this log message to say:
`"Authorization code not present. Authorization code generated successfully."`